### PR TITLE
docs(website): update docs for dependencies

### LINKED ISSUE #280: Website: Add documentation of optional dependencies to the "Writing a Feature App" guide

### PR DESCRIPTION
**Pull request by KingHenne**
fixes #280

### CODE REVIEW

File: docs/guides/writing-a-feature-app.md [hunk 1]
```diff
@@ -37,8 +49,33 @@ required Feature Services][feature-service-binder] to the dependent Feature App.
...
+   declared. If one of those dependencies can't be fulfilled, the Feature App
+   won't be created. This means the Feature App can be sure that those
+   dependencies are always present when it is created.
+
+   Feature Service dependencies are declared with their ID as key, and a [semver
+   version range][semver] as value, e.g.
+   `{'acme:some-feature-service': '^2.0.0'}`.
+
+1. With `dependencies.externals` all required external dependencies are
+   declared. This may include [shared npm
```
Comment (clebert) [inline docs/guides/writing-a-feature-app.md]: Who could declare externals other than the integrator?

File: docs/guides/writing-a-feature-app.md [hunk 2]
```diff
@@ -37,8 +49,33 @@ required Feature Services][feature-service-binder] to the dependent Feature App.
...
+   dependencies][sharing-npm-dependencies] that are provided by the integrator.
+
+   External dependencies are declared with their module name as key, and a
+   [semver version range][semver] as value, e.g. `{react: '^16.7.0'}`.
+
+## `optionalDependencies`
```
Comment (clebert) [inline docs/guides/writing-a-feature-app.md]: Perhaps a sentence would be good that Externals currently can not yet be declared as optional. But there is already an issue which can be tackled on demand.

File: docs/guides/writing-a-feature-app.md (hunk 1 above)
Comment (fahrradflucht) [reply docs/guides/writing-a-feature-app.md]: We remembered, that we also had the possibility in mind that the `ExternalsValidator` is used to version other "externals" besides npm packages, like APIs exposed on the `window` or stuff like that.

File: docs/guides/writing-a-feature-app.md (hunk 2 above)
Comment (KingHenne) [reply docs/guides/writing-a-feature-app.md]: done

**Pull request merged**